### PR TITLE
Tmedia 743 small promo block

### PR DESCRIPTION
**Pull request by brentmiller1973**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Updated the small-promo-block to work under AS system/

## Jira Ticket

- [[TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-743)

## Acceptance Criteria

Small Promo block is migrated to the new architecture.

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-743-small-promo-block`
2. Run `npm run storybook`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/small-promo-block`

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/small-promo-block/package.json [hunk 1]
```diff
@@ -20,15 +19,9 @@
...
-  "scripts": {
-    "test": "echo \"Error: run tests from root\" && exit 1",
-    "lint": "eslint --ext js --ext jsx features"
-  },
   "peerDependencies": {
     "@arc-fusion/prop-types": "^0.1.5",
-    "@wpmedia/engine-theme-sdk": "*",
-    "@wpmedia/news-theme-css": "*",
-    "@wpmedia/shared-styles": "*"
+    "@wpmedia/engine-theme-sdk": "*"
```
Comment (matthewroach) [inline blocks/small-promo-block/package.json:25]:
```suggestion
    "@wpmedia/arc-themes-components: "*",
    "@wpmedia/engine-theme-sdk": "*"
```

File: blocks/small-promo-block/package.json [hunk 2]
```diff
@@ -2,14 +2,13 @@
   "name": "@wpmedia/small-promo-block",
   "version": "5.17.0",
   "description": "Fusion News Theme Small Promo block",
```
Comment (matthewroach) [inline blocks/small-promo-block/package.json]:
```suggestion
  "description": "Small Promo block",
```

File: .storybook/themes/news.scss [hunk 3]
```diff
@@ -1207,6 +1207,31 @@
...
+					grid-template-columns: 1fr,
+					components: (
+						heading: (
+							font-size: var(--body-font-size),
+							line-height: var(--body-line-height),
+						),
+					),
+				),
+				small-promo-above: (
+					grid-template-columns: 1fr,
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: This is redundant as it's already set on the `small-promo`

File: .storybook/themes/news.scss [hunk 4]
```diff
@@ -1207,6 +1207,31 @@
...
+				),
+				small-promo-below: (
+					grid-template-columns: 1fr,
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: This is redundant as it's already set on the `small-promo`

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/small-promo-block/features/small-promo/default.jsx [hunk 5]
```diff
@@ -20,70 +36,123 @@ const SmallPromoItem = ({ customFields, arcSite }) => {
...
 	if (shouldLazyLoad && isServerSide()) {
 		return null;
 	}
+
+	const linkURL = content?.websites?.[arcSite]?.website_url;
+	const imageURL =
+		content.promo_items?.basic?.type === "image" && content.promo_items?.basic?.url
+			? content.promo_items.basic.url
+			: null;
+	const headline = content?.description?.basic;
```
Comment (matthewroach) [inline blocks/small-promo-block/features/small-promo/default.jsx]: This should be the headline not the description

File: blocks/small-promo-block/features/small-promo/default.jsx [hunk 6]
```diff
@@ -20,70 +36,123 @@ const SmallPromoItem = ({ customFields, arcSite }) => {
...
-	customFields = { showImage: true, showHeadline: true, imageRatio: "3:2" },
-}) => {
-	const { isAdmin, arcSite } = useFusionContext();
-	const shouldLazyLoad = customFields?.lazyLoad && !isAdmin;
 	if (shouldLazyLoad && isServerSide()) {
 		return null;
 	}
+
+	const linkURL = content?.websites?.[arcSite]?.website_url;
+	const imageURL =
```
Comment (matthewroach) [inline blocks/small-promo-block/features/small-promo/default.jsx]: We should use the util function from Arc Themes Components to get the image `getImageFromANS`

File: .storybook/themes/news.scss (hunk 3 above)
Comment (brentmiller1973) [reply .storybook/themes/news.scss]: I removed the declaration

File: .storybook/themes/news.scss (hunk 4 above)
Comment (brentmiller1973) [reply .storybook/themes/news.scss]: I removed the declaration

File: blocks/small-promo-block/features/small-promo/default.jsx (hunk 5 above)
Comment (brentmiller1973) [reply blocks/small-promo-block/features/small-promo/default.jsx]: Changed to `const headline = content?.headlines?.basic;`

File: blocks/small-promo-block/features/small-promo/default.jsx (hunk 6 above)
Comment (brentmiller1973) [reply blocks/small-promo-block/features/small-promo/default.jsx]: Changed to `const imageURL = getImageFromANS(content);`

**Review by matthewroach — APPROVED** [review]
Looks good, development approved - passing to @haganka for design QA

**Comment by nschubach** [comment]
We should be deleting code from the `shared-styles` as these last promos are being converted.  (Should be able to delete `shared-styles/_children/promos/small` with this?)

File: blocks/small-promo-block/features/small-promo/default.jsx [hunk 7]
```diff
@@ -20,70 +37,120 @@ const SmallPromoItem = ({ customFields, arcSite }) => {
...
+	const PromoHeading = () =>
+		showHeadline && headline ? (
+			<Heading>
+				{linkURL ? (
+					<Link href={formatURL(linkURL)} onClick={registerSuccessEvent}>
+						{headline}
+					</Link>
+				) : (
+					headline
+				)}
```
Comment (nschubach) [inline blocks/small-promo-block/features/small-promo/default.jsx]:
```suggestion
				<Conditional component={Link} condition={linkURL} href={formatURL(linkURL)} onClick={registerSuccessEvent}>
					{headline}
				</Conditional>
```

File: blocks/small-promo-block/features/small-promo/default.jsx [hunk 8]
```diff
@@ -20,70 +37,120 @@ const SmallPromoItem = ({ customFields, arcSite }) => {
...
+				</MediaItem>
+			) : null;
+		return showImage && linkURL ? (
+			<Link href={formatURL(linkURL)} onClick={registerSuccessEvent} assistiveHidden>
+				{ImageDisplay}
+			</Link>
+		) : (
+			ImageDisplay
+		);
+	};
```
Comment (nschubach) [inline blocks/small-promo-block/features/small-promo/default.jsx]:
```suggestion
	const PromoImage = () => {
		const { fallbackImage } = getProperties(arcSite);
		const availableImage = imageURL || fallbackImage;
		return (showImage && availableImage) ? (
			<Conditional component={Link} condition={linkURL} onClick={registerSuccessEvent} assistiveHidden>
				<MediaItem>
					<Image alt="" src={availableImage} />
				</MediaItem>			
			</Conditional>
		) : null;
	};
```

**Comment by matthewroach** [comment]
@visionbegin We will need to update the headline as per @nschubach's suggestions to address the issue we are seeing in the other promo's during testing

**Comment by brentmiller1973** [comment]
> We should be deleting code from the `shared-styles` as these last promos are being converted. (Should be able to delete `shared-styles/_children/promos/small` with this?)

This has been nuked :-)

**Comment by haganka** [comment]
Hey @visionbegin I'm seeing some funky errors in Chromatic. Could you take a peek?

Screen Shot 2022-07-14 at 3 56 02 PM

Screen Shot 2022-07-14 at 3 55 52 PM

**Comment by brentmiller1973** [comment]
> Hey @visionbegin I'm seeing some funky errors in Chromatic. Could you take a peek?
> 
> Screen Shot 2022-07-14 at 3 56 02 PM
> 
> Screen Shot 2022-07-14 at 3 55 52 PM

@haganka You should be good to go now!

**Pull request merged**